spec/Section 2 -- Language.md

```diff
@@ -938,3 +943,239 @@ and operations.
+system definitions should return a descriptive error.
+
+Note: This IDL is used throughout the remainder of this specification document
+when illustrating example type systems.
+
+
+### Schema Definition
+
+SchemaDefinition : schema { OperationTypeDefinition+ }
+
+OperationTypeDefinition : OperationType : NamedType
+
+A GraphQL Type System includes exactly one Schema Definition, which defines
+the type to be used for each operation.
+
+In this example, a GraphQL schema is defined with both query and mutation types:
+
+```graphql
+schema {
+  query: QueryRoot
+  mutation: MutationRoot
+}
+
+type QueryRoot {
+  someField: String
+}
+
+type MutationRoot {
+  setSomeField(to: String): String
+}
+```
+
+
+### Type Definition
+
+TypeDefinition :
+  - ScalarTypeDefinition
+  - ObjectTypeDefinition
+  - InterfaceTypeDefinition
+  - UnionTypeDefinition
+  - EnumTypeDefinition
+  - InputObjectTypeDefinition
+
+A GraphQL Type System is defined by many different kinds of types.
+
+
+#### Scalar
+
+ScalarTypeDefinition : scalar Name
+
+Scalar types represent leaf values in a GraphQL type system. While this GraphQL
+specification describes a set of Scalar types which all GraphQL services must
+supply, custom Scalar types may be supplied by a GraphQL service. Typically, the
+set of Scalar types described in this specification are omitted from documents
+which describe a schema for brevity.
+
+In this example, a Scalar type called `DateTime` is defined:
+
+```graphql
+scalar DateTime
+```
+
+
+#### Object
+
+ObjectTypeDefinition : type Name ImplementsInterfaces? { FieldDefinition+ }
+
+ImplementsInterfaces : implements NamedType+
+
+FieldDefinition : Name ArgumentsDefinition? : Type
+
+ArgumentsDefinition : ( InputValueDefinition+ )
+
+InputValueDefinition : Name : Type DefaultValue?
+
+Object types represent a list of named fields, each of which yield a value of a
+specific type. Each field itself may accept a list of named arguments.
+
+Objects may implement Interface types. When implementing an Interface type, the
+Object type must supply all fields defined by the Interface.
+
+In this example, a Object type called `TodoItem` is defined:
+
+```graphql
+type TodoItem implements Node {
+  id: ID
+  title: String
+  isCompleted: Boolean
+}
+```
+
+#### Interface
+
+InterfaceTypeDefinition : interface Name { FieldDefinition+ }
+
+Interface types, similarly to Object types represent a list of named fields.
+Interface types are used as the type of a field when one of many possible Object
+types may yielded during execution, but some fields are guaranteed. An Object
+type is a possible type of an Interface when it both declares that it implements
+that Interface as well as includes all defined fields.
+
+In this example, an Interface type called `Node` is defined:
+
+```graphql
+interface Node {
+  id: ID
+}
+```
+
+#### Union
+
+UnionTypeDefinition : union Name = UnionMembers
+
+UnionMembers :
```

Comment:
Oh, the `|` is meant to be taken literally. That solves an apparent grammatical ambiguity too.